app/com/arpnetworking/metrics/portal/alerts/impl/PluggableAlertRepository.java

```diff
@@ -119,24 +126,38 @@ public PluggableAlertRepository(
      * @param periodicMetrics A metrics instance to record against.
      * @param configProvider The config loader for the alert definitions.
      * @param org The organization to group the alerts under.
+     * @param openTimeout The timeout to use when waiting for the first update at open.
      */
     public PluggableAlertRepository(
             final ObjectMapper objectMapper,
             final PeriodicMetrics periodicMetrics,
             final ConfigProvider configProvider,
-            final UUID org
-    ) {
+            final UUID org,
+            final Duration openTimeout
+            ) {
         _objectMapper = objectMapper;
         _configProvider = configProvider;
         _organization = new DefaultOrganization.Builder().setId(org).build();
         _periodicMetrics = periodicMetrics;
+        _openTimeout = openTimeout;
     }
 
     @Override
     public void open() {
         assertIsOpen(false);
         LOGGER.debug().setMessage("Opening PluggableAlertRepository").log();
-        _configProvider.start(this::reload);
+        final CompletableFuture<Void> initialReload = new CompletableFuture<>();
+        _configProvider.start(stream -> {
+            reload(stream);
+            if (!initialReload.isDone()) {
+                initialReload.complete(null);
+            }
+        });
+        try {
+            initialReload.get(_openTimeout.toMillis(), TimeUnit.MILLISECONDS);
+        } catch (final InterruptedException | ExecutionException | TimeoutException e) {
+            throw new RuntimeException("failed waiting for initial reload", e);
```

Comment:
I'm unsure if I should re-throw here since it will likely prevent startup instead of just breaking alerting.